sway/commands/move.c

```diff
@@ -0,0 +1,181 @@
+	};
+	for (size_t i = 0; i < sizeof(names) / sizeof(names[0]); ++i) {
+		if (strcasecmp(names[i].name, direction) == 0) {
+			struct wlr_output *adjacent = wlr_output_layout_adjacent_output(
+					root_container.sway_root->output_layout,
+					names[i].direction, reference, ref_lx, ref_ly);
+			if (adjacent) {
+				struct sway_output *sway_output = adjacent->data;
+				return sway_output->swayc;
+			}
+			break;
+		}
+	}
+	return output_by_name(direction);
+}
+
+static struct cmd_results *cmd_move_container(struct sway_container *current,
+		int argc, char **argv) {
+	struct cmd_results *error = NULL;
+	if ((error = checkarg(argc, "move container/window",
+				EXPECTED_AT_LEAST, 4))) {
+		return error;
+	} else if (strcasecmp(argv[1], "to") == 0
+			&& strcasecmp(argv[2], "workspace") == 0) {
+		// move container to workspace x
+		if (current->type == C_WORKSPACE) {
+			// TODO: Wrap children in a container and move that
+			return cmd_results_new(CMD_FAILURE, "move", "Unimplemented");
+		} else if (current->type != C_CONTAINER && current->type != C_VIEW) {
+			return cmd_results_new(CMD_FAILURE, "move",
+					"Can only move containers and views.");
+		}
+		struct sway_container *ws;
+		const char *num_name = NULL;
+		char *ws_name = NULL;
+		if (argc == 5 && strcasecmp(argv[3], "number") == 0) {
+			// move "container to workspace number x"
+			num_name = argv[4];
+			ws = workspace_by_number(num_name);
+		} else {
+			ws_name = join_args(argv + 3, argc - 3);
+			ws = workspace_by_name(ws_name);
+		}
+		if (!ws) {
+			ws = workspace_create(ws_name ? ws_name : num_name);
+		}
+		free(ws_name);
+		struct sway_container *old_parent = current->parent;
+		struct sway_container *focus = sway_seat_get_focus_inactive(
+				config->handler_context.seat, ws);
+		container_move_to(current, focus);
+		sway_seat_set_focus(config->handler_context.seat, old_parent);
+		return cmd_results_new(CMD_SUCCESS, NULL, NULL);
+	} else if (strcasecmp(argv[1], "to") == 0
+			&& strcasecmp(argv[2], "output") == 0) {
+		if (current->type == C_WORKSPACE) {
+			// TODO: Wrap children in a container and move that
+			return cmd_results_new(CMD_FAILURE, "move", "Unimplemented");
+		} else if (current->type != C_CONTAINER
+				&& current->type != C_VIEW) {
+			return cmd_results_new(CMD_FAILURE, "move",
+					"Can only move containers and views.");
+		}
+		struct sway_container *source = container_parent(current, C_OUTPUT);
+		struct sway_container *destination = output_in_direction(argv[3],
+				source->sway_output->wlr_output, current->x, current->y);
+		if (!destination) {
+			return cmd_results_new(CMD_FAILURE, "move workspace",
+				"Can't find output with name/direction '%s'", argv[3]);
+		}
+		struct sway_container *focus = sway_seat_get_focus_inactive(
+				config->handler_context.seat, destination);
+		if (!focus) {
+			// We've never been to this output before
+			focus = destination->children->items[0];
+		}
+		container_move_to(current, focus);
```

Comment:
`sway_seat_set_focus`?

Reply:
Maybe not. In which case, should probably remove line 84.

Reply:
No, I think we need to do this here. i3 doesn't bring focus along for the ride.

Reply:
Though I have always wanted an option to make moves follow, cause it's nice for moving a bunch of things at once